pyproject.toml

```diff
@@ -30,7 +30,10 @@ boto3 = "^1.24.11"
 botocore = "^1.27.11"
 pandas = "^1.2.0, <=1.5.1, !=1.5.0" # Exclusion per: https://github.com/aws/aws-sdk-pandas/issues/1678
 numpy = "^1.22.4"
-pyarrow = ">=6.0.0, <8.1.0"
+pyarrow = [
+    { version = ">=6.0.1, <8.0.0", markers = "extra == 'ray' or 'ray' in extra" },
```

Comment:
Even though `in` is defined by the [grammar](https://peps.python.org/pep-0508/#complete-grammar) I couldn't quite make it work for the combinations of extras.. Just putting it out there

Reply:
Is this saying that we should only install `pyarrow` when we use `ray`? Don't we use `pyarrow` in non-distributed `read_parquet` as well?

Reply:
There's 2nd line below that that covers non-ray

Reply:
🤦 my bad!

It looks like the static check build is still failing with:
> ModuleNotFoundError: No module named 'pyarrow'